### ISSUE TITLE
Add load from checkpoint option

### ISSUE THREAD

**Issue by felix-e-h-p** [openclimatefix/pvnet#534]
Maybe be good to have as two PRs accounting for just carrying on RE crashing and another for fine tuning.

- So one for a full state resume.
- Second for a weight only warm start.

**Comment by lm2612** [comment]
Agree, will also add a test to check this works and that model weights match hyperparameters set in the config file.
1. For full state resume, we should check that training continues from same rng, learning rate state, etc. 
2. For fine-tuning from a warm start - we may be providing different config files for datamodule, training, and optimizer part of model.

**Comment by felix-e-h-p** [comment]
> Agree, will also add a test to check this works and that model weights match hyperparameters set in the config file.
> 
> 1. For full state resume, we should check that training continues from same rng, learning rate state, etc.
> 2. For fine-tuning from a warm start - we may be providing different config files for datamodule, training, and optimizer part of model.

Yeah, for the first perhaps just key in the config and then in train():

 trainer.fit(model=model, datamodule=datamodule, ckpt_path=ckpt_path)

Or around that.

And for second maybe a new function in load.py and from that just a new section in existing config.

**Comment by lm2612** [comment]
Thanks Felix, I am looking at adding `ckpt_path` to config as you suggest. While looking into a test for it, I noticed that the directory path for the model checkpoints isn't being used properly:

https://github.com/openclimatefix/pvnet/blob/8bb428c852da22db56b6f577d1a2ea76c8c1339c/configs.example/callbacks/default.yaml#L27-L28

I started adding a new test in the branch [resume_from_ckpt_path](https://github.com/openclimatefix/pvnet/blob/resume_from_ckpt_path/tests/training/test_train.py) that should test if we can resume training from the checkpoint saved after running`test_train_pvnet`. I was expecting checkpoints to be saved into `str(Path(wandb_save_dir).parent / "ckpts"`. But they are being saved in `Path(wandb_save_dir).parent/runid`.

Anyone know if this is to be expected? Do we need to remove the `dirpath` option if it isn't being used?

**Comment by felix-e-h-p** [comment]
> Thanks Felix, I am looking at adding `ckpt_path` to config as you suggest. While looking into a test for it, I noticed that the directory path for the model checkpoints isn't being used properly:
> 
> [pvnet/configs.example/callbacks/default.yaml](https://github.com/openclimatefix/pvnet/blob/8bb428c852da22db56b6f577d1a2ea76c8c1339c/configs.example/callbacks/default.yaml#L27-L28)
> 
> Lines 27 to 28 in [8bb428c](/openclimatefix/pvnet/commit/8bb428c852da22db56b6f577d1a2ea76c8c1339c)
> 
>  # The path to where the model checkpoints will be stored 
>  dirpath: "PLACEHOLDER/${model_name}" #${..model_name} 
> I started adding a new test in the branch [resume_from_ckpt_path](https://github.com/openclimatefix/pvnet/blob/resume_from_ckpt_path/tests/training/test_train.py) that should test if we can resume training from the checkpoint saved after running`test_train_pvnet`. I was expecting checkpoints to be saved into `str(Path(wandb_save_dir).parent / "ckpts"`. But they are being saved in `Path(wandb_save_dir).parent/runid`.
> 
> Anyone know if this is to be expected? Do we need to remove the `dirpath` option if it isn't being used?

Yeah dirpath doesn't resolve so Lightning falls back to wandb save dir. The test just misses it because it points at its own path rather than the configured dirpath.

I'd fix it rather than remove it. Like `${hydra:runtime.output_dir}` and add a test there asserting a checkpoint. Can do it as a small PR before other updates I suppose.

**Comment by lm2612** [comment]
Ok I see what you mean, dirpath is replaced with the wandb path [here](https://github.com/openclimatefix/pvnet/blob/8bb428c852da22db56b6f577d1a2ea76c8c1339c/pvnet/training/train.py#L97)
I can add that in the test instead. 
I'd also suggest a comment in the config.example saying that dirpath won't be used if using wandb (if that's correct)
